modules/alphamat/tutorials/alphamat_tutorial.markdown

```diff
@@ -0,0 +1,53 @@
+Information Flow Alpha Matting {#tutorial_alphamat}
+============================
+
+This project was part of Google Summer of Code 2019.
+
+*Student:* Muskaan Kularia
+
+*Mentor:* Sunita Nayak
+
+Alphamatting is the problem of extracting the foreground from an image. The extracted foreground can be used for further operations like changing the background in an image.
+
+Given an input image and its corresponding trimap, we try to extract the foreground from the background. Following is an example:
+
+Input Image: ![](samples/input_images/plant.jpg)
+Input Trimap: ![](samples/trimaps/plant.png)
+Output alpha Matte: ![](samples/output_mattes/plant_result.jpg)
+
+This project is implementation of @cite aksoy2017designing . It required implementation of parts of other papers [2,3,4].
+
+# Building
+
+This module uses the Eigen package.
+
+Build the sample code of the alphamat module using the following cmake command, followed by  `make` command.
+```
+cmake -DOPENCV_EXTRA_MODULES_PATH=<path to opencv_contrib modules> -DBUILD_EXAMPLES=ON ..
+
+make
```

Comment:
> make

`cmake --build . --config Release --target example_alphamat_information_flow_matting` ?

Reply:
alphamat module needs the core and imgproc modules too. Do you have an example of doing it using the cmake --target way you mentioned above?

Reply:
Build process handles required dependencies **automatically**.

> example

Use `opencv_core` / `opencv_imgproc`, but you don't need it here.

Reply:
Checked it works. Updated the instructions.